crates/sui-core/src/authority/execution_time_estimator.rs

```diff
@@ -74,16 +76,19 @@ impl ExecutionTimeObserver {
         let mut observer = Self {
             epoch_store: Arc::downgrade(epoch_store),
             consensus_adapter,
-            local_observations: HashMap::new(),
+            local_observations: LruCache::new(
+                NonZero::new(lru_cache_size).expect("lru_cache_size must be non-zero"),
+            ),
         };
-        tokio::spawn(async move {
+        let epoch_store = epoch_store.clone();
```

Comment:
I'd change the function arg here also to just be the by-value copy you need instead of taking a & and then cloning inside the function